litgpt/pretrain.py

```diff
@@ -333,21 +333,24 @@ def fit(
 
 @torch.no_grad()
 def validate(fabric: L.Fabric, model: nn.Module, val_dataloader: DataLoader, max_iters: int) -> torch.Tensor:
+    fabric.barrier()
```

Comment:
The barrier here and at the end are for sanity, they are not necessary for the fix.

Reply:
Not a fan of the overzealous barriers if they don't fix anything specific.

Reply:
They don't cost anything and I would say it's good practice when switching loops.